barretenberg/cpp/src/barretenberg/vm/avm/trace/execution.hpp

```diff
@@ -27,18 +28,15 @@ class Execution {
     static VmPublicInputs convert_public_inputs(std::vector<FF> const& public_inputs_vec);
 
     // TODO: Clean these overloaded functions. We probably need less and confusing overloading.
-    static std::vector<Row> gen_trace(std::vector<Instruction> const& instructions,
-                                      std::vector<FF>& returndata,
+    static std::vector<Row> gen_trace(std::vector<FF>& returndata,
                                       std::vector<FF> const& calldata,
                                       std::vector<FF> const& public_inputs,
-                                      ExecutionHints const& execution_hints = {});
-    static std::vector<Row> gen_trace(std::vector<Instruction> const& instructions,
-                                      std::vector<FF> const& calldata = {},
-                                      std::vector<FF> const& public_inputs = {});
-    static std::vector<Row> gen_trace(std::vector<Instruction> const& instructions,
-                                      std::vector<FF> const& calldata,
-                                      std::vector<FF> const& public_inputs,
-                                      ExecutionHints const& execution_hints);
+                                      ExecutionHints const& execution_hints,
+                                      std::vector<uint8_t> const& bytecode);
+
+    static std::vector<Row> gen_trace(std::vector<FF> const& calldata = {},
+                                      std::vector<FF> const& public_inputs = {},
+                                      std::vector<uint8_t> const& bytecode = {});
```

Comment:
It's ok if you want to put it at the end, but you likely never want an empty bytecode. You may need to just put it at the beginning.

You've already gone the lazy way with the hack for `returndata` and never came back to fix it! 😆 I don't forget